modules/shinezBidAdapter.js

```diff
@@ -1,435 +1,435 @@
+  return !isEmpty(val) ? encodeURIComponent(val) : '';
+}
+
+/**
+ * Get preferred user-sync method based on publisher configuration
+ * @param bidderCode {string}
+ * @returns {string}
+ */
+function getAllowedSyncMethod(filterSettings, bidderCode) {
+  const iframeConfigsToCheck = ['all', 'iframe'];
+  const pixelConfigToCheck = 'image';
+  if (filterSettings && iframeConfigsToCheck.some(config => isSyncMethodAllowed(filterSettings[config], bidderCode))) {
+    return SUPPORTED_SYNC_METHODS.IFRAME;
+  }
+  if (!filterSettings || !filterSettings[pixelConfigToCheck] || isSyncMethodAllowed(filterSettings[pixelConfigToCheck], bidderCode)) {
+    return SUPPORTED_SYNC_METHODS.PIXEL;
+  }
+}
+
+/**
+ * Check if sync rule is supported
+ * @param syncRule {Object}
+ * @param bidderCode {string}
+ * @returns {boolean}
+ */
+function isSyncMethodAllowed(syncRule, bidderCode) {
+  if (!syncRule) {
+    return false;
+  }
+  const isInclude = syncRule.filter === 'include';
+  const bidders = isArray(syncRule.bidders) ? syncRule.bidders : [bidderCode];
+  return isInclude && contains(bidders, bidderCode);
+}
+
+/**
+ * Get the seller endpoint
+ * @param testMode {boolean}
+ * @returns {string}
+ */
+function getEndpoint(testMode) {
+  return testMode
+    ? SELLER_ENDPOINT + MODES.TEST
+    : SELLER_ENDPOINT + MODES.PRODUCTION;
+}
+
+/**
+ * get device type
+ * @param uad {ua}
+ * @returns {string}
+ */
+function getDeviceType(ua) {
+  if (/ipad|android 3.0|xoom|sch-i800|playbook|tablet|kindle/i
+    .test(ua.toLowerCase())) {
+    return '5';
+  }
+  if (/iphone|ipod|android|blackberry|opera|mini|windows\sce|palm|smartphone|iemobile/i
+    .test(ua.toLowerCase())) {
+    return '4';
+  }
+  if (/smart[-_\s]?tv|hbbtv|appletv|googletv|hdmi|netcast|viera|nettv|roku|\bdtv\b|sonydtv|inettvbrowser|\btv\b/i
+    .test(ua.toLowerCase())) {
+    return '3';
+  }
+  return '1';
+}
+
+function generateBidsParams(validBidRequests, bidderRequest) {
+  const bidsArray = [];
+
+  if (validBidRequests.length) {
+    validBidRequests.forEach(bid => {
+      bidsArray.push(generateBidParameters(bid, bidderRequest));
+    });
+  }
+
+  return bidsArray;
+}
+
+/**
+ * Generate bid specific parameters
+ * @param {bid} bid
+ * @param {bidderRequest} bidderRequest
+ * @returns {Object} bid specific params object
+ */
+function generateBidParameters(bid, bidderRequest) {
+  const {params} = bid;
+  const mediaType = isBanner(bid) ? BANNER : VIDEO;
+  const sizesArray = getSizesArray(bid, mediaType);
+
+  // fix floor price in case of NAN
+  if (isNaN(params.floorPrice)) {
+    params.floorPrice = 0;
+  }
+
+  const bidObject = {
+    mediaType,
+    adUnitCode: getBidIdParameter('adUnitCode', bid),
+    sizes: sizesArray,
+    floorPrice: Math.max(getFloor(bid, mediaType), params.floorPrice),
+    bidId: getBidIdParameter('bidId', bid),
+    bidderRequestId: getBidIdParameter('bidderRequestId', bid),
+    transactionId: bid.ortb2Imp?.ext?.tid || '',
```

Comment:
curious why github seeing the large diff on the file

Reply:
I don't know either, using git the diff is one line as it's supposed to be. let's see what happens on my next update